src/db/emoji_logging.rs

```diff
@@ -25,34 +20,56 @@ impl EmojiStats {
     }
 }
 
+macro_rules! query_ordering {
+    ($conn:expr,$query:expr,$limit:tt) => {{
+        let records = sqlx::query!($query, $limit).fetch_all(&mut $conn).await?;
+
+        Ok(records
+            .into_iter()
+            .map(|x| EmojiStats {
+                emoji: EmojiIdentifier {
+                    id: EmojiId(x.emoji_id as u64),
+                    animated: x.animated != 0,
+                    name: x.emoji_name.unwrap(),
+                },
+                in_text: x.in_text_usage as u64,
+                reactions: x.reaction_usage as u64,
+            })
+            .collect())
+    }};
+}
```

Comment:
This definitely needs a doc-comment, including an explanation as to _why_ it exists. Additionally, it should probably be named sth like "process_emoji_stats_query" - as this macro itself is not concerned with anything related to the ordering, it's just necessary because of it